knotx-core/knotx-service-adapter/src/main/java/com/cognifide/knotx/core/serviceadapter/ServiceAdapterResponse.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Knot.x - Sample App with Mock service
+ *
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.knotx.core.serviceadapter;
+
+import java.util.Map;
+
+import io.vertx.core.json.JsonObject;
+
+public class ServiceAdapterResponse {
+
+  private Map<String, Object> serviceResult;
+
+  private boolean success;
+
+  private String reason;
+
+  private ServiceAdapterResponse() {
+    // hidden default constructor
+  }
+
+
+  public ServiceAdapterResponse(JsonObject body) {
+    this.success = body.getBoolean("success");
+    if (success) {
+      // TODO deserialize service result
+    } else {
+      this.reason = body.getString("reason");
+    }
+  }
+
+  public static ServiceAdapterResponse success(Map<String, Object> serviceResult) {
```

Comment:
To be honest we could get rid of that class at all, instead:
- Use HttpResponseWrapper (it have body filed where response will be available - serialization you will get for free, and if you wanted to get body as Json - you can create method to return new JsonObject(body) if body is json of course)
- you'd have all the information we'd need now and leter (status code, headers..)
- on the engine level that will use that response will be a proper logic based on headers, status codes etc. that let you know make a decision whether to process the response or not, or to get the actual result object..